packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/components/LogoModalStepper/index.js

```diff
@@ -1,59 +1,81 @@
-import React, { useReducer, useEffect } from 'react';
+import React, { useReducer } from 'react';
 import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 import { ModalLayout, ModalHeader } from '@strapi/design-system/ModalLayout';
 import { Typography } from '@strapi/design-system/Typography';
 import reducer, { initialState } from './reducer';
-import stepper from './stepper';
 
-const LogoModalStepper = ({ initialStep, isOpen, onClose }) => {
-  const [{ currentStep }, dispatch] = useReducer(reducer, initialState);
-  const { Component, modalTitle, next } = stepper[currentStep];
+const LogoModalStepper = ({
+  onChangeLogo,
+  customLogo,
+  goTo,
+  Component,
+  modalTitle,
+  next,
+  prev,
+  currentStep,
+}) => {
+  const [{ localImage }, dispatch] = useReducer(reducer, initialState);
   const { formatMessage } = useIntl();
 
-  useEffect(() => {
-    if (isOpen) {
-      goTo(initialStep);
-    }
-    // Disabling the rule because we just want to open the modal at a specific step
-    // then we let the stepper handle the navigation
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isOpen]);
-
-  const goTo = to => {
-    dispatch({
-      type: 'GO_TO',
-      to,
-    });
-  };
-
   const setLocalImage = asset => {
     dispatch({
       type: 'SET_LOCAL_IMAGE',
       value: asset,
     });
   };
 
-  if (!isOpen) {
+  if (!currentStep) {
     return null;
   }
 
   return (
-    <ModalLayout labelledBy="modal" onClose={onClose}>
+    <ModalLayout labelledBy="modal" onClose={() => goTo(null)}>
       <ModalHeader>
         <Typography fontWeight="bold" as="h2" id="modal">
           {formatMessage(modalTitle)}
         </Typography>
       </ModalHeader>
-      <Component setLocalImage={setLocalImage} goTo={goTo} next={next} onClose={onClose} />
+      <Component
+        setLocalImage={setLocalImage}
+        goTo={goTo}
+        next={next}
+        prev={prev}
+        onClose={() => goTo(null)}
```

Comment:
Could you create a function for the onClose event so you can also use it in the ModalLayout please?